File: 238. Product of Array Except Self/Solution.py
class Solution:
    def productExceptSelf(self, nums: List[int]) -> List[int]:
        zeros = nums.count(0)
        if zeros>1:
            return [0]*len(nums)
        ind = None
        prod = 1
        arr = [0]*len(nums)
        for i in range(len(nums)):
            if nums[i]==0:
                ind = i
            else:
                prod*=nums[i]
        if ind is not None:
            arr[ind] = prod
        else:
            for i in range(len(nums)):
                arr[i] = prod//nums[i]
        return arr

File: 238. Product of Array Except Self/Solution.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var productExceptSelf = function(nums) {
    let countZero = ((arr)=>{
        let count = 0;
        for(let i=0;i<arr.length;i++){
            if (arr[i]==0){
                count++;
            }
        }
        return count;
    });
    let zeros = countZero(nums);
    if (zeros > 1){
        return new Array(nums.length).fill(0);
    }
    let ind = null;
    let prod = 1;
    let arr = new Array(nums.length).fill(0);
    for(let i = 0;i<nums.length;i++){
        if(nums[i]==0){
            ind = i;
        }else{
            prod *= nums[i];
        }
    }
    if(ind!=null){
        arr[ind] = prod;
    }else{
        for(let i=0;i<nums.length;i++){
            arr[i] = prod/nums[i];
        }
    }
    return arr;
};
